README.md

```diff
@@ -67,6 +68,14 @@ func main() {
   	// Enable BLE interface.
 	must("enable BLE stack", adapter.Enable())
 
+	ctx, cancel := context.WithCancel(context.Background())
+	adapter.SetConnectHandler(func(device bluetooth.Device, connected bool){
```

Comment:
Actually, perhaps you want to change the entire handler to this:

```go
	adapter.SetConnectHandler(func(device bluetooth.Device, connected bool) {
		if connected {
			println("device connected:", device.Address.String())
			return
		}

		println("device disconnected:", device.Address.String())
		cancel()
	})
```

I think that perhaps better captures your intent here? Please let me know what you think.